src/H5VLcallback.c

```diff
@@ -3424,11 +3429,24 @@ H5VL__file_open_find_connector_cb(H5PL_type_t plugin_type, const void *plugin_in
     }
     H5E_END_TRY;
 
-    /* If the file was accessible with the current VOL connector, return
-     * the FAPL with that VOL connector set on it. Errors are ignored here
-     * as some VOL connectors may not support H5Fis_accessible.
-     */
-    if (status == SUCCEED && is_accessible > 0) {
+    if (status < 0) {
+        ssize_t new_num_errors = 0;
+
+        /* Pop any errors generated by the above call */
+        if ((new_num_errors = H5Eget_num(H5E_DEFAULT)) < 0)
+            HGOTO_ERROR(H5E_ERROR, H5E_CANTGET, H5_ITER_ERROR, "can't get current error stack size")
+        if (new_num_errors > num_errors) {
+            new_num_errors -= num_errors;
+            if (H5Epop(H5E_DEFAULT, (size_t)new_num_errors) < 0)
+                HGOTO_ERROR(H5E_ERROR, H5E_CANTRELEASE, H5_ITER_ERROR, "can't sanitize error stack")
```

Comment:
An example of using H5Epop to remove expected error messages generated from the H5E_BEGIN/END_TRY block above so they don't get included in the eventual error stack printing.